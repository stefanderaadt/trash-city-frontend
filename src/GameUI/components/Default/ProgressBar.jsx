import React from 'react';

const ProgressBar = ({ progress, description }) => {
  return (
    <div className="progress-bar">
      <div className="progress-bar-display" style={{ width: `${progress}%` }} />
      <div className="progress-bar-text">{description ? description(progress) : progress}</div>
    </div>
  );
};

export default ProgressBar;
