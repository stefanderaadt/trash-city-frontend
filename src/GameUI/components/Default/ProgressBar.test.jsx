import React from 'react';
import { shallow } from 'enzyme';

import ProgressBar from './ProgressBar';

describe('ProgressBar tests', () => {
  test('ProgressBar component rendering as expected', () => {
    const wrapper = shallow(<ProgressBar progress={78} />);
    expect(wrapper).toMatchSnapshot();
  });
});
