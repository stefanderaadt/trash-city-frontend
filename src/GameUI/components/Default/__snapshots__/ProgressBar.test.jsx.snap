// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar tests ProgressBar component rendering as expected 1`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar-display"
    style={
      Object {
        "width": "78%",
      }
    }
  />
  <div
    className="progress-bar-text"
  >
    78
  </div>
</div>
`;
